### DISCUSSION TITLE
When to use provider.notifier

### DISCUSSION THREAD

**Discussion by sbergmair** [rrousselGit/riverpod#1210] [Q&A]
When should I use `myChangeNotifierProvider.notifier` and when only `myChangeNotifierProvider`.
I figured out, that `ref.watch(myChangeNotifierProvider.notifier)` doesn't work, because it doesn't get add to the listeners of the ChaneNotifier. 
Is there a ruling how to use each of this types?

**Comment by TimWhiting** [comment]
`.notifier` notifies whenever a new `ChangeNotifier` is created (likely because the `ChangeNotifierProvider` is watching another provider that changes).
Without the `.notifier` you get notified anytime the `ChangeNotifier` calls `notifiyListeners()`

**Comment by rrousselGit** [comment]
This is covered by the .notifier documentation https://pub.dev/documentation/flutter_riverpod/latest/flutter_riverpod/AutoDisposeChangeNotifierProvider/notifier.html
